src/status_im/contexts/wallet/send/events.cljs

```diff
@@ -116,6 +116,44 @@
  (fn [{:keys [db]}]
    {:db (update-in db [:wallet :ui :send] dissoc :amount)}))
 
+(rf/reg-event-fx
+ :wallet/open-send-flow
+ (fn [{:keys [db]} [{:keys [recipient address stack-id]}]]
+   (let [[_ address-without-prefix] (utils/split-prefix-and-address address)
+         wallet-accounts            (vals (get-in db [:wallet :accounts]))
+         default-account-address    (some #(when (:default-account? %) (:address %))
+                                          wallet-accounts)
+         multiple-accounts?         (-> (filter :operable? wallet-accounts)
+                                        count
+                                        (> 1))]
+     {:db (cond-> db
+            (not multiple-accounts?)
+            (assoc-in [:wallet :current-viewing-account-address] default-account-address)
+
+            :always
+            (update-in
+             [:wallet :ui :send]
+             assoc
+             :recipient  (or recipient address)
+             :to-address address-without-prefix))
+      :fx [[:dispatch [:hide-bottom-sheet]]
+           [:dispatch [:shell/change-tab :wallet-stack]]
+           [:dispatch [:pop-to-root :shell-stack]]
+           [:dispatch-later
+            [{:ms       600
+              :dispatch (if multiple-accounts?
+                          [:open-modal :screen/wallet.select-from]
+                          [:wallet/wizard-navigate-forward
+                           {:current-screen stack-id
+                            :start-flow?    true
+                            :flow-id        :wallet-send-flow}])}]]]})))
+
+(rf/reg-event-fx
+ :wallet/init-send-flow-for-address
+ (fn [_ [args]]
+   {:fx [[:dispatch [:wallet/clean-send-data]]
+         [:dispatch [:wallet/open-send-flow args]]]}))
```

Comment:
We clean the send data before user goes into the flow. This ensure no data from any other flows is left and also to prevent any bugs.

Reply:
Is `:wallet/open-send-flow` the only way the send flow is started? If so, we could include the cleaning once in there, since it seems to be required every time we start the flow.

Reply:
No @clauxx, as far as i checked, we have other entry points which we need to cover. We can unify the event into this which we need to explore later.